include/commons.hpp

```diff
@@ -30,64 +30,76 @@
 #include <stdlib.h>
 
-#include <bit>
-#include <span>
-#include <thread>
-#include <functional>
 #include <algorithm>
+#include <bit>
 #include <cassert>
 #include <chrono>
 #include <cstddef>
 #include <cstdint>
 #include <execution>
+#include <functional>
 #include <iostream>
 #include <iterator>
+#include <map>
 #include <memory>
 #include <numeric>
+#include <span>
+#include <thread>
 #include <type_traits>
 #include <typeinfo>
 #include <vector>
 
 #include "counting_iterator.hpp"
 
 // get mdpsan 2d indices from 1d index
-#define dim2(x, ms)          \
-  int ii = x / ms.extent(1); \
-  int ij = x % ms.extent(1);
+#define dim2(x, ms)            \
+    int ii = x / ms.extent(1); \
+    int ij = x % ms.extent(1);
 // get mdspan 3d indices from 1d index
-#define dim3(x, ms)                            \
-  int ii = x / (ms3.extent(1) * ms.extent(2)); \
-  int ij = (x / ms.extent(2)) % ms.extent(1);  \
-  int ik = x % ms.extent(2)
+#define dim3(x, ms)                              \
+    int ii = x / (ms3.extent(1) * ms.extent(2)); \
+    int ij = (x / ms.extent(2)) % ms.extent(1);  \
+    int ik = x % ms.extent(2)
 
 class Timer {
- public:
-  Timer() { start(); }
+   public:
+    Timer() { start(); }
 
-  ~Timer() { stop(); }
+    ~Timer() { stop(); }
 
-  void start() { start_time_point = std::chrono::high_resolution_clock::now(); }
+    void start() { start_time_point = std::chrono::high_resolution_clock::now(); }
 
-  double stop() {
-    end_time_point = std::chrono::high_resolution_clock::now();
-    return duration();
-  }
+    double stop() {
+        end_time_point = std::chrono::high_resolution_clock::now();
+        return duration();
+    }
 
-  double duration() {
-    auto start = std::chrono::time_point_cast<std::chrono::microseconds>(
-                     start_time_point)
-                     .time_since_epoch()
-                     .count();
-    auto end =
-        std::chrono::time_point_cast<std::chrono::microseconds>(end_time_point)
-            .time_since_epoch()
-            .count();
-    auto duration = end - start;
-    double ms = duration * 0.001;
-    return ms;
-  }
+    double duration() {
+        auto start =
+            std::chrono::time_point_cast<std::chrono::microseconds>(start_time_point).time_since_epoch().count();
+        auto end = std::chrono::time_point_cast<std::chrono::microseconds>(end_time_point).time_since_epoch().count();
+        auto duration = end - start;
+        double ms = duration * 0.001;
+        return ms;
+    }
 
- private:
-  std::chrono::time_point<std::chrono::high_resolution_clock> start_time_point;
-  std::chrono::time_point<std::chrono::high_resolution_clock> end_time_point;
+   private:
+    std::chrono::time_point<std::chrono::high_resolution_clock> start_time_point;
+    std::chrono::time_point<std::chrono::high_resolution_clock> end_time_point;
 };
+
+enum class sch_t { CPU, GPU, MULTIGPU };
+
+[[nodiscard]] sch_t get_sch_t(std::string_view str) {
```

Comment:
Let's rename this function to `get_sch_enum()` or something similar as `_t` suffix is used for type aliases so it may be confusing for reader to determine if this is a type or function?

Reply:
good point!